src/OpenConext/ProfileBundle/Controller/LocaleController.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * Copyright 2015 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace OpenConext\ProfileBundle\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+
+class LocaleController extends Controller
+{
+    public function switchLocaleAction(Request $request)
+    {
+        $returnUrl = $request->query->get('return-url');
+
+        // Return URLs generated by us always include a path (ie. at least a forward slash)
+        // @see https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpFoundation/Request.php#L878
+        $domain = $request->getSchemeAndHttpHost() . '/';
+
+        if (strpos($returnUrl, $domain) !== 0) {
+            $this->get('logger')->error(sprintf(
+                'Illegal return-url for redirection after changing locale, aborting request'
+            ));
+            throw new BadRequestHttpException('Invalid return-url given');
+        }
+
+        $form = $this->createForm(
+            'profile_switch_locale',
+            null,
+            [
+                'current_locale' => $request->getLocale(),
+                'route_parameters' => ['return_url' => $returnUrl]
```

Comment:
Since we're using a static route now (`profile.locale_switch_locale`), the required parameters are known. Just a `return_url` option rather than a `route_parameters` option is clearer, simpler. Agreed?